wcomponents-theme/src/test/intern/wc.dom.getLabelsForElement.test.js

```diff
@@ -66,6 +66,45 @@ define(["intern!object", "intern/chai!assert", "./resources/test.utils!"],
 				var result = controller(document.getElementById("unlabelled"));
 				assert.isTrue(Array.isArray(result));
 				assert.strictEqual(0, result.length);
+			},
+			testAriaLabel: function() {
+				var element = document.getElementById("aria-text");
+				var labels = controller(element);
+				var expected = document.getElementById("aria-label");
+				assert.strictEqual(expected, labels[0]);
+			},
+			testAriaLabelUnavailable: function() {
+				var element = document.getElementById("aria-text2");
+				var labels = controller(element);
+				assert.strictEqual(0, labels.length);
+			},
+			testAriaMultipleLabel: function() {
+				var element = document.getElementById("aria-text3");
+				var labels = controller(element);
+				var expected1 = document.getElementById("aria-label3");
+				var expected2 = document.getElementById("aria-label4");
+				assert.strictEqual(2, labels.length);
+				assert.strictEqual(expected1, labels[0]);
+				assert.strictEqual(expected2, labels[1]);
+			},
+			testAriaMultipleLabel2: function() {
+				var element = document.getElementById("aria-input");
+				var labels = controller(element);
+				var expected1 = document.getElementById("billing");
+				var expected2 = document.getElementById("name");
+				assert.strictEqual(2, labels.length);
+				assert.strictEqual(expected1, labels[0]);
+				assert.strictEqual(expected2, labels[1]);
+			},
+			testDivElementAriaLabel: function() {
+				var element = document.getElementById("main");
+				var labels = controller(element);
+				var expected = document.getElementById("foo");
+				assert.strictEqual(expected, labels[0]);
+			},
+			testNullElement: function() {
+				var labels = controller(null);
+				assert.strictEqual(0, labels.length);
 			}
 		});
 	});
```

Comment:
Good tests ✨ 
There are still some important cases missing, for example what happens if an element is labelled using native semantics (e.g. nested in a label or via the `for` attribute) as well as ARIA semantics? Which takes precedence? Does it find them all?

Reply:
You mean like this? If yes it is already covered.
> input type="radio" name="sex" id="male" aria-labelledby="aria-label"

Reply:
@kandula-aswin that sample would only be relevant to this question if it also has a native label associated with it.

Reply:
There are two native ways to label a form element:
```html
<!-- Nesting -->
<label>Surname<input/></label>

<!-- idref (for attribute) -->
<label for="surname">Surname</label>
<input id="surname"/>
```

Ideally only one technique should be applied and ideally it should be native semantics.
In practice it is *possible* to combine ARIA techniques with native techniques.
In our discussions and whiteboarding we decided that, in this case, the native labels should be the first items in the returned array. I do not see that reflected in the tests, did I miss it?

Reply:
Ok yes I see now, you got it already.